### PR TITLE
Prepare for release 1.2.0-rc4

### PR DESCRIPTION
**Pull request by yophilav**
# 1.2.0-rc4 (2020-03-1)
## AWARENESS
This release contains a significant refactoring to the IoT Edge security daemon. It separates out the daemon's functionality for provisioning and providing cryptographic services for Linux-based devices into a set of stand-alone system services. Details on these individual system services can be found in the [Overview](https://azure.github.io/iot-identity-service) of the related github repository in which they reside. 

### Impact to Edge modules
Every attempt has been made to ensure that the APIs on which Edge modules depend will remain unaffected and backward compatible. Issues affecting Edge modules will be treated with the highest priority.

### Impact to installing / configuring IoT Edge
The refactoring does affect the packaging and installation of IoT Edge. While we've attempted to minimize the impact of these there are expected differences. For more details on these changes please refer to the discussion of [Packaging](https://azure.github.io/iot-identity-service/packaging.html).

### Bug Fixes
* Improve Edge Agent's Prometheus parser ( [a4ae2c1](https://github.com/Azure/iotedge/commit/a4ae2c116296ad086a9f2cb7ccd2f077c5692301) )
* Update Base Images for Security Vulnerability ( [6fe4de3](https://github.com/Azure/iotedge/commit/6fe4de3f16a11d7a294a278b55a07b15fdf7ecd6) )
* Consolidate environment variables for EdgeHub and Broker ( [6167323](https://github.com/Azure/iotedge/commit/6167323e772a61d55f2120c9673360194dd4f1b4) )
* Fix getmodules error filter rfc3339 datetime ( [3a98e83](https://github.com/Azure/iotedge/commit/3a98e830d762317325b77ba8b4958bc0d878c06c) )
* Update current config when a plan is empty ( [087de0b](https://github.com/Azure/iotedge/commit/087de0b59d3354bf17055ff91371bd024494db48) )

### Bug Fixes
* Fix MQTT client persistent session's subscription when the client reconnects ( [7e2d191](https://github.com/Azure/iotedge/commit/7e2d191cb8b4cd45bcec7653c523b8f8ab5e707f) )
* Consolidate log levels between EdgeHub and Broker ( [45bcc83](https://github.com/Azure/iotedge/commit/45bcc835b00b940923a99839de4548cdbbabc619) )
* Restrict local authentication for ApiProxy connections to port 8080 ( [11bfea5](https://github.com/Azure/iotedge/commit/11bfea5fb35bf1f1a98ec6ff402343d32bd8dde4) )
* Update Base Images for Security Vulnerability ( [6fe4de3](https://github.com/Azure/iotedge/commit/6fe4de3f16a11d7a294a278b55a07b15fdf7ecd6) )
* Stop EdgeHub if initial twin has no base configuration ( [7701250](https://github.com/Azure/iotedge/commit/770125073da574cff7249bb46328443aa899e58a) )
* Fix Message Store Exception when cleaning up an updated message ( [a478ce8](https://github.com/Azure/iotedge/commit/a478ce84630fc59398aa53969e684d0f01905673) )
* Disable the use of http proxy for Amqp/Mqtt over TCP ( [e3f9c27](https://github.com/Azure/iotedge/commit/e3f9c27a2613e0e5181b97225a42e4986207eaa9) )
* Added timeout sending messages to MQTT broker to prevent infinite waiting ( [0ce632f](https://github.com/Azure/iotedge/commit/0ce632f675d092c7867c0c77d3a1239a0abd29d0) )
* Improve EdgeHub config parsing ( [a59cfe2](https://github.com/Azure/iotedge/commit/a59cfe233c1ef7cc807d60a4678500ecb76ab282) )
* Set default auth mode to Scope ( [399a9a3](https://github.com/Azure/iotedge/commit/399a9a357e2dd98b03bd46ab81d303dd12bb0bdb) )

### Features
* Optimize Nested Edge descendents mapping ( [2cf92db](https://github.com/Azure/iotedge/commit/2cf92dbe26e5e256239613678f3b3f84f8b94935) )
* Support MQTT bridge events ( [a31eed4](https://github.com/Azure/iotedge/commit/a31eed41c60d8b18ecd3b524e3235a8978c219e1) )
* Handle thumbprint auth through API proxy for WebSocket ( [f710e43](https://github.com/Azure/iotedge/commit/f710e439afee72edc561486e2a0ef6d9d0994bfa) )
* Introduce unsubscription through nested levels on client-disconnect ( [eca18ee](https://github.com/Azure/iotedge/commit/eca18eec732409f80237026ee6a81e4dfc5c42b6) )

### Bug Fixes
* 1.50 Rust toolchain update ( [040c54d](https://github.com/Azure/iotedge/commit/040c54dedf2f29a33c3302b468379b82c5dd3276) )
* Adding web sockect support for API proxy  ( [86ab1a0](https://github.com/Azure/iotedge/commit/86ab1a0ec158916410281744b7b64c005d6644e4) )
* Update Base Images for Security Vulnerability ( [6edad21](https://github.com/Azure/iotedge/commit/6edad2143d37af8a4213ffbe898e873faec36b58) )
* Fixing error message at API proxy start ( [4629a74](https://github.com/Azure/iotedge/commit/4629a749ef19adcca289eb573d1c2ec6e259bbbf) )
* Allow EdgeAgent image to be resolved via parent address ( [32834a6](https://github.com/Azure/iotedge/commit/32834a662a014365b26be13a86756368f2f3a5bf) )
* Change API proxy to accept a trustbundle with more than one root certificate ( [f1fc6d9](https://github.com/Azure/iotedge/commit/f1fc6d9c092985b4f5d395c5effce03f4e5cacd1) )

### Features
* Handle thumbprint auth through API proxy for WebSocket ( [f710e43](https://github.com/Azure/iotedge/commit/f710e439afee72edc561486e2a0ef6d9d0994bfa) )

### Bug Fixes
* 1.50 Rust toolchain update ( [040c54d](https://github.com/Azure/iotedge/commit/040c54dedf2f29a33c3302b468379b82c5dd3276) )
* Persist in-flight queue on broker restart ( [6d81b94](https://github.com/Azure/iotedge/commit/6d81b94d413484e41243469e2430bff75ae514ee) )
* Send MQTT "will" on broker shutdown ( [3dced1c](https://github.com/Azure/iotedge/commit/3dced1ce9bfcc7fdb9b15348dbe6d6e152503175) )
* Retry non-iothub subscription when rejected by server ( [be3d482](https://github.com/Azure/iotedge/commit/be3d482d522394b00db740c127173c23549a31dc) )

### Features
* Introduced Ring Buffer to MQTT Bridge ( [354b04f](https://github.com/Azure/iotedge/commit/354b04fd2b29bf26702f1cf637bcb8357a98582d), [a24f6bb](https://github.com/Azure/iotedge/commit/a24f6bb27221a75993f92e1f40af49279ca77643), [e9d4d2b](https://github.com/Azure/iotedge/commit/e9d4d2b1f3f46389dacd97977cc8975635f67ab5), [c9d7ea3](https://github.com/Azure/iotedge/commit/c9d7ea3456b2b8c7893a050a5d1779d0f59a14de), [b86f014](https://github.com/Azure/iotedge/commit/b86f01494796cb7698d0c0eab9ed718755982c10), [8df7e88](https://github.com/Azure/iotedge/commit/8df7e8806712844d8e3ef53bcd41b01d1d37f09a), [2af9dac](https://github.com/Azure/iotedge/commit/2af9dacdc499f2f4326b5ef896d68df38ad3e43d), [6f12c88](https://github.com/Azure/iotedge/commit/6f12c88b6c1d9227f95e9f12c2c496083426415f), [a027808](https://github.com/Azure/iotedge/commit/a027808352d6c7f971d810235411a0fd956f4653), [7b35db9](https://github.com/Azure/iotedge/commit/7b35db925862162820999086c415879b264edbcc) )
* Configuration improvement ( [45bcc83](https://github.com/Azure/iotedge/commit/45bcc835b00b940923a99839de4548cdbbabc619), [a8223af](https://github.com/Azure/iotedge/commit/a8223af8c437ac5155e1c9b2b8341b734773ab5b) )

### Bug Fixes
* Improve `iotedge check` ( [6c7fc9b](https://github.com/Azure/iotedge/commit/6c7fc9b75360f34f9d66db6dc1d2656fa04fd83c), [e254d9c](https://github.com/Azure/iotedge/commit/e254d9cabbcb07ad7247bffa68082c4c6dd8b45c), [7245c8e](https://github.com/Azure/iotedge/commit/7245c8e053b97596dd05a6c70160d4569f77bee4), [455cef9](https://github.com/Azure/iotedge/commit/455cef92f2945cbe08c5b5e5abb39fe5311353ff) )
* Bugfix iotedge CLI ( [e254d9c](https://github.com/Azure/iotedge/commit/e254d9cabbcb07ad7247bffa68082c4c6dd8b45c), [c6a9bbb](https://github.com/Azure/iotedge/commit/c6a9bbb44737bac51c8945ed86bf776ef9a8279a) )
* Fix import of listen URIs when original config used socket activation ( [e4794ae](https://github.com/Azure/iotedge/commit/e4794aee0aa6c7d78bee83d62662d5935bf8f381) )
* Configuration changes for content trust with certificate service ( [9e2f4b8](https://github.com/Azure/iotedge/commit/9e2f4b8b03cc75358ad2a44dc6a213299115721e) )
* Retry getting device information on startup ( [492a159](https://github.com/Azure/iotedge/commit/492a15918aa2facaf130b7993ace663cd5ffe077) )
* Add iotedge-init-import command to migrate pre-1.2 config to 1.2+ config ( [e3bf3c9](https://github.com/Azure/iotedge/commit/e3bf3c9871bc9339253ac662eeefbcae43782287) )
* Temporarily raise quickstart EdgeCA cert's expiry to 30 days ( [9cc5b8f](https://github.com/Azure/iotedge/commit/9cc5b8f727bcd7de51fb021c8b231aadf8d4f381) )

### Features
* Convert iotedged config to TOML, and implement `iotedge config` ( [e254d9c](https://github.com/Azure/iotedge/commit/e254d9cabbcb07ad7247bffa68082c4c6dd8b45c), [d0978bf](https://github.com/Azure/iotedge/commit/d0978bf63bdd5624543680424452ee5c08fe285a) )
* Support Nested Edge topology ( [e254d9c](https://github.com/Azure/iotedge/commit/e254d9cabbcb07ad7247bffa68082c4c6dd8b45c), [dc7c929](https://github.com/Azure/iotedge/commit/dc7c92944beb3747c6f5341321025c9b541056f6) )
* Introduce `iotedge system` commands ( [cbe03af](https://github.com/Azure/iotedge/commit/cbe03af512105fcc3b8899a5aebce547cf924de9) )
* Make edgelet's certificate CA configurable ( [6073a78](https://github.com/Azure/iotedge/commit/6073a78a44af6dbdb78a28ac5b473b0b0b2d2874) )

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,159 @@
+# 1.2.0-rc4 (2020-03-1)
+## BREAKING CHANGE 
+This release uses new Identity Service (IS) artifact which is not backward compatible.
```
Comment (varunpuranik) [inline CHANGELOG.md:3]: @micahl - can you pls sign off on the wording of the doc here? Should we add more details on what this entails?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,159 @@
...
+* Add message cleanup interval environment variable ([dbcc2d9](https://github.com/Azure/iotedge/commit/dbcc2d91afeca8a21ba170f6ae595b1f4d1cd645))
+* Update service SDK to 1.28.1-NestedEdge and and devices SDK to 1.33.1-NestedEdge ([858106f](https://github.com/Azure/iotedge/commit/858106ffcb90c435c396eef39fad29cd355de3cb))
+* Support for sha256 thumbprint authentication ([dac5710](https://github.com/Azure/iotedge/commit/dac5710d79b034031ead89c14a66feb04ed5d8a7))
+* Add client connection related EdgeHub metrics ([c7da97b](https://github.com/Azure/iotedge/commit/c7da97bf1a2df6b7995613631dc8258cd8b6aa4b))
+
+## Other Module Images
+### Bug Fixes
+* Update arm base images for security vulnerability ([07f6750](https://github.com/Azure/iotedge/commit/07f6750958063adc92f5c732fb6a8eca8a9a1dea))
+
+# 1.2.0-rc2 (2020-11-20)
```
Comment (varunpuranik) [inline CHANGELOG.md:111]: Why are we including these changes right now?

**Review by varunpuranik — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (yophilav) [reply CHANGELOG.md:111]: There seems to be an overwrite of the history in release/1.2 branch. 
I manually port over these changes from release page
https://github.com/Azure/iotedge/commit/f87761c623cd19cd717d6019c237e331a19730fe

**Review by varunpuranik — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (yophilav) [reply CHANGELOG.md:3]: I have confirmed w/ Micah on the terminology here :)

Comment (micahl) [reply CHANGELOG.md:3]: @yophilav let's use the below to introduce the RC4...  And then I'll also go make sure the content in those links is in sync with the current state.

**AWARENESS**
This release contains a significant refactoring to the IoT Edge security daemon. It separates out the functionality for provisioning and providing cryptographic services for Linux-based devices into a set of stand-alone system services. Details on these individual system services can be found in the [Overview](https://azure.github.io/iot-identity-service) of the related github repository in which they reside. 

_Impact to Edge modules_
Every attempt has been made to ensure that the APIs on which Edge modules depend will remain unaffected and backward compatible. Issues affecting Edge modules will be treated with the highest priority. 

_Impact to installing / configuring IoT Edge_
The refactoring does affect the packaging, installation, and configuration of IoT Edge's host-level components. While we've attempted to minimize the impact of these, there are expected differences. For more details on these changes please refer to the discussion of [Packaging](https://azure.github.io/iot-identity-service/packaging.html).

**Review by micahl — APPROVED** [review]

**Pull request merged**